.github/workflows/continuous-integration.yml

```diff
@@ -19,9 +19,26 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: inspect binary file sizes
+      - name: detect changed files
+        run: |
+          INTERESTING_FILES=$(
+            sort -u < <(
+              # Changes that are not committed (staged + unstaged + untracked), but without deleted files
+              git status --short -v -v --no-renames --porcelain | awk '$1 != "D" { print $2 }'
+              # Changes committed so far (may overlap with the above)
+              git diff --no-renames --name-only --diff-filter=d ${{ github.sha }}^
+            )
+          )
+
+          # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+          DELIMITER="X${RANDOM}"
```

Comment:
Where's `${RANDOM}` coming from here? I don't see it set anywhere, and I don't see it on the list of github-provided envvars...

Reply:
it's provided by the shell, an excerpt from `man bash` below:

```
       RANDOM Each  time this parameter is referenced, it expands to a random integer between 0 and 32767.  Assigning
              a value to RANDOM initializes (seeds) the sequence of random numbers.  If RANDOM is unset, it loses its
              special properties, even if it is subsequently reset.
```